MinecraftClient/Protocol/Handlers/Protocol18.cs

```diff
@@ -413,8 +417,26 @@ internal bool HandlePacket(int packetID, Queue<byte> packetData)
                             {
                                 if (protocolversion >= MC114Version)
                                     dataTypes.ReadNextNbt(packetData);  // Heightmaps - 1.14 and above
+                                int biomesLength = 0;
+                                if (protocolversion >= MC1162Version)
+                                    if (chunksContinuous)
+                                        biomesLength = dataTypes.ReadNextVarInt(packetData); // Biomes length - 1.16.2 and above
                                 if (protocolversion >= MC115Version && chunksContinuous)
-                                    dataTypes.ReadData(1024 * 4, packetData); // Biomes - 1.15 and above
+                                {
+                                    if (protocolversion >= MC1162Version)
+                                    {
+                                        for (int i = 0; i < biomesLength; i++)
+                                        {
+                                            // Biomes - 1.16.2 and above
+                                            // Don't use ReadNextVarInt because it cost too much time
+                                            while (true)
+                                            {
+                                                if ((dataTypes.ReadNextByte(packetData) & 0x80) != 128) break;

```

Comment:
Clever! Seems worth to move into a dedicated method `dataTypes.SkipNextVarInt(packetData);` 👍